okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientStream.java

```diff
@@ -96,6 +96,10 @@ public void transportHeadersReceived(List<Header> headers, boolean endOfStream)
     synchronized (lock) {
       if (endOfStream) {
         transportTrailersReceived(Utils.convertTrailers(headers));
+        if (getTransportError() != null) {
```

Comment:
Yes, remove the endOfStream condition in transport*Received. In some cases it will be unnecessary, but it won't hurt anything. Since this is an error case, I'm not too concerned with making it optimal; I'm mostly concerned with making it correct and maintainable.